mp/src/game/client/momentum/tf2proxy.cpp

```diff
@@ -45,7 +45,20 @@ class CInvulnLevelProxy : public CResultProxy
 {
   public:
     bool Init(IMaterial *pMaterial, KeyValues *pKeyValues) { return true; }
-    void OnBind(void *pC_BaseEntity) {}
+    void OnBind(void *pC_BaseEntity)
+    {
+        Assert(m_pResult);
+
+        if (!pC_BaseEntity)
+            return;
+
+        C_BaseEntity *pEntity = BindArgToEntity(pC_BaseEntity);
+        if (!pEntity)
+            return;
```

Comment:
Is this needed? The entity isn't actually used in the OnBind function, do we need to cast it? I guess this also applies to the yellow level one too.

Reply:
When I added the YellowLevel proxy, this was part of the code snippet ficool sent me, I thought I'd add these checks just in case it would cause issues otherwise, but if it's not needed, I can remove it from both proxies.

Reply:
I would check if it's needed by removing it and seeing if it still works.

Reply:
I am not able to check on Windows right now, but it appears that the game crashes without fail on Linux for me when I start an sj map with the TF2 sticky launcher model script, whether I keep the checks or not.

This is what is shown in the terminal at the time of the crash:
```
IDirect3DDevice9::CreatePixelShader: shaderapi's centroid mask (0x00000000) differs from mask derived from shader name (0x0000000C) for shader ps-file flashlight_ps20b ps-index 32 ps-combo 0
IDirect3DDevice9::CreatePixelShader: shaderapi's centroid mask (0x00000000) differs from mask derived from shader name (0x0000000C) for shader ps-file flashlight_ps20b ps-index 32 ps-combo 1
IDirect3DDevice9::CreatePixelShader: shaderapi's centroid mask (0x00000000) differs from mask derived from shader name (0x0000000C) for shader ps-file flashlight_ps20b ps-index 32 ps-combo 2
IDirect3DDevice9::CreatePixelShader: shaderapi's centroid mask (0x00000000) differs from mask derived from shader name (0x0000000C) for shader ps-file flashlight_ps20b ps-index 32 ps-combo 3
```

Reply:
Does this happen on steam build too? Or just on this branch?

Reply:
Only on this branch.

Reply:
What about latest develop?

Reply:
Latest develop works fine too.

The crash only happens as soon as I add `m_pResult->SetFloatValue(0.0f);` to the proxy.

Reply:
Hmm. Maybe just `#ifdef` it out for now then. Weird that it's causing this, and yeah I'd fix this to be under the right proxy.